src/Ouroboros-Mini_Protocols-Chain_Sync.thy

```diff
@@ -0,0 +1,178 @@
+section \<open>Chain Synchronization Mini-Protocol\<close>
+
+text \<open>
+  The theory in this section implements the chain synchronization mini-protocol.
+\<close>
+
+theory "Ouroboros-Mini_Protocols-Chain_Sync"
+  imports
+    "Ouroboros-Mini_Protocols"
+    "HOL-Library.BNF_Corec"
+begin
+
+locale chain_sync =
+  fixes initial_items\<^sub>c :: "'i list"
+  fixes initial_items\<^sub>s :: "'i list"
+  fixes candidate_points :: "'i list \<Rightarrow> 'p list"
+  fixes best_intersection_point :: "'i list \<Rightarrow> 'p list \<Rightarrow> 'p option"
```

Comment:
Fixed in 2065ea0890c42f9d88a26cfd4da149a89a0a070b, please check.

Reply:
What about changing `set (map ψ 𝒞)` in the definition of `first_intersection_point` to ``ψ ` set 𝒞``? It’s more “mathematical” (sets instead of lists) and more compact.

Reply:
Fixed in 8bfebf133ad45620d70fe3e11238960d3d70ed1f, please check.